src/test/java/pages/UserProfilePage.java

```diff
@@ -0,0 +1,413 @@
+		fieldFirstName.clear();
+		fieldFirstName.sendKeys(Fname);
+		fieldLastname.clear();
+		fieldLastname.sendKeys(Lname);
+		fieldPhoneNumber.clear();
+		fieldPhoneNumber.sendKeys(PhoneNumber);
+		fieldAddress.clear();
+		fieldAddress.sendKeys(Address);
+		fieldCountry.clear();
+		fieldCountry.sendKeys(Country);
+		fieldState.clear();
+	    fieldState.sendKeys(State);
+	    fieldCity.clear();
+		fieldCity.sendKeys(City);
+		fieldpincode.clear();
+		fieldpincode.sendKeys(Pincode);
+		
+		}
+	public String getUpdateMessageforPersonalDetails() {
+		String Msg = profileUpdateMessage.getText();
+		return Msg;
+	}
+	
+// -----Update Password------------------------------------------------
+	
+@FindBy (xpath = "//*[@formcontrolname='currentPassword']")
+public WebElement currentPassword;
+
+@FindBy (xpath = "//*[@formcontrolname='newPassword']")
+public WebElement newPassword;
+
+@FindBy (xpath = "//*[@formcontrolname='newConfirmPassword']")
+public WebElement newConfirmPassword;
+
+@FindBy (xpath = "//*[@class='error-msg'][contains(text(),'Current Password must have atleast 8 characters')]")
+public WebElement errorMessageCurrentPassword;
+
+@FindBy (xpath = "//*[@class='error-msg'][contains(text(),'New Password must have atleast 8 characters')]")
+public WebElement errorMessageNewPassword;
+
+@FindBy (xpath = "//*[@class='error-msg'][contains(text(),'Confirm Password must have atleast 8 characters')]")
+public WebElement errorMessageConfirmPassword;
+
+@FindBy (xpath = "//*[@class='error-msg'][contains(text(),'New password and Confirm password must be same')]")
+public WebElement errorMessageNewPasswordMismatch;
+
+@FindBy (xpath = "//*[@class='error-msg'][contains(text(),'Current password and New password must be different')]")
+public WebElement errorMessageSameCurrentandNewPassword;
+
+
+@FindBy (xpath ="//*[@role=\"alertdialog\"][contains(text(),'The current password does not match with your existing password')]")
+public WebElement errorMessageIncorrectCurrentPassword;
+
+
+
+public String getIncorrectCurrentPasswordMessage() {
+	String Msg = errorMessageIncorrectCurrentPassword.getText();
+	return Msg;
+}
+
+public String getNewPasswordMismatchMessage() {
+	String Msg = errorMessageNewPasswordMismatch.getText();
+	return Msg;
+}
+
+public String getSameNewandCurrentPasswordMessage() {
+	String Msg = errorMessageSameCurrentandNewPassword.getText();
+	return Msg;
+}
+
+public void clearFields() {
+ currentPassword.clear();
+ newPassword.clear();
+ newConfirmPassword.clear();
+}
+
+
+@FindBy (xpath ="//*[@class='btn btn-primary '][contains(text(),'Save')]")
+public WebElement savePasswordBtn;
+
+@FindBy (xpath ="//*[@class='btn btn-primary'][contains(text(),'Cancel')]")
+public WebElement CancelPasswordBtn;
+
+//-------Number of Games Played----------------------------------------
+	
+	@FindBy (xpath = "//*[contains(@src,\"test-icons.png\")]/..//span[@class=\"bignum\"]")
+	public WebElement gamesCount;
+	
+	public boolean fetchCountOfGamesPlayed(){
```

Comment:
What is this method being used?

Reply:
fetchCountOfGamesPlayed() is being used for fetching the no of games played in Edit Profile Screen